src/main/java/org/lsc/Hooks.java

```diff
@@ -63,31 +64,56 @@ public class Hooks {
 	 *
 	 * return nothing
 	 */
-	public final static void postSyncHook(final String hook, final LscModifications lm) {
+	public final static void postSyncHook(final String hook, final String outputFormat, final LscModifications lm) {
 
 		if( hook != null && ! hook.equals("") )
 		{
-			// Compute json modifications
-			String jsonModifications = null;
+
+			String format = "";
+			if( outputFormat.equals("json") ) {
+				format = "json";
+			}
+			else
+			{
+				format = "ldif";
+			}
+
+			// Compute json/ldif modifications
+			String modifications = null;
 
 			switch (lm.getOperation()) {
 				case CREATE_OBJECT:
-					jsonModifications = getJsonModifications(lm);
-					callHook("create", hook, lm.getMainIdentifier(), jsonModifications);
+					if( format.equals("json") ) {
+						modifications = getJsonModifications(lm);
+					}
+					else {
+						modifications = LdifLayout.format(lm);
+					}
+					callHook("create", hook, lm.getMainIdentifier(), format, modifications);
```

Comment:
these lines can be easily factorized, and in this case easily integrated in the callHook method